crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1149,14 +1149,27 @@ impl World {
     /// });
     /// assert_eq!(world.get_resource::<A>().unwrap().0, 2);
     /// ```
-    pub fn resource_scope<R: Resource, U>(&mut self, f: impl FnOnce(&mut World, Mut<R>) -> U) -> U {
+    pub fn resource_scope<
+        R: 'static, /* The resource doesn't need to be Send nor Sync. */
```

Comment:
Really we should remove the `Send/Sync` supertraits on resources, because the fact that you can add anything as a non send resource is a bit messy.

But this is fine for now.

Reply:
I think you should open another issue about it, since it's out of the scope of this PR.